File: ignisign-android/app/src/main/java/com/ignisign/example/ContractContext.kt
package com.ignisign.example

import com.google.gson.Gson
import com.ignisign.ignisign.IgnisignApplicationEnv

data class ContractContext(
    val signatureRequestId: String,
    val ignisignSignerId: String,
    val ignisignSignatureToken: String,
    val ignisignUserAuthSecret: String,
    val ignisignAppId: String,
    val ignisignAppEnv: IgnisignApplicationEnv,
    val signerEmail: String? = null,
    val documentId: String? = null
) {
    fun jsonToContractContext(json: String): ContractContext? {
        return try {
            Gson().fromJson(json, ContractContext::class.java)
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }

    override fun toString(): String {
        return "ContractContext(signatureRequestId='$signatureRequestId', ignisignSignerId='$ignisignSignerId', ignisignSignatureToken='$ignisignSignatureToken', ignisignUserAuthSecret='$ignisignUserAuthSecret', ignisignAppId='$ignisignAppId', ignisignAppEnv=$ignisignAppEnv, signerEmail=$signerEmail, documentId=$documentId)"
    }


}

File: ignisign-ios/Example/ignisign-ios/ContractContext.swift
//
//  ContractContext.swift
//  ignisign-ios_Example
//
//  Created by Marc Nigdélian on 22/12/2023.
//  Copyright © 2023 CocoaPods. All rights reserved.
//

import Foundation
import ignisign_ios

struct ContractContext: Decodable {
    let signatureRequestId: String
    let ignisignSignerId: String
    let ignisignSignatureToken: String
    let ignisignUserAuthSecret: String
    let ignisignAppId: String
    let ignisignAppEnv: IgnisignApplicationEnv
    let signerEmail: String?
    let documentId: String?
}
